p2p/net/session_cache.go

```diff
@@ -0,0 +1,90 @@
+package net
+
+import (
+	"errors"
+	"github.com/spacemeshos/go-spacemesh/p2p/p2pcrypto"
+	"sync"
+	"time"
+)
+
+// simple cache for storing sessions
+const maxSessions = 1024
+
+type storedSession struct {
```

Comment:
In fact they are quite similar, connectionPool is more complicated because it has to synchronize long dials. also connection pool sessions are cached by the PublicKey received in session key exchange. for UDP we can't do this unless we attach the pubkey to each message.